docs/src/best_practices.md

```diff
@@ -12,4 +12,32 @@ From the perspective of software, we can break the neatly into two components. T
 
 `Simulation Util` instead, as discussed above, wraps around open-source libraries and algorithmic routines that aids in the simulation of the Navier-Stokes equations. `Simulation Constructs` and `Simulation Util` are divided in a way that source files in `Simulation Util` would only be dependent on `Simulation Constructs` but not vis versa.
 
-## Testing
\ No newline at end of file
+## Testing
+
+Here, we introduce the testing methodology for software, based on the `Test.jl` package in Julia's standard library. As software gets developed, modifications to the codebase would inevitably change the behaviour of outputs. By placing in safeguards in terms of automated testing (since it is software that can be run automatically by a microprocessor), bugs introduced during the development can be avoided and flagged to the developer in a manner appropriate. The automated testing of the codebase is part of the continuous integration (CI) software development practice, that we would touch upon below.
+
+In general, testing can be broken into levels of granularity. At the lowest level of granularity, we have unit tests, suitably written for testing the small chunks of code, generally functions and basis properties of structures. Climbing one level up, we have integration testing. Here, the focus is more on validation of correct software behaviour between inputs and outputs of functions. As software written gets complicated, systems level testing is introduced as the next level of granularity, which now is considered a black-box technique, usually tasked to be executed by software quality engineers instead of developers and hence we omit ourselves from the systems testing and higher level granularity at this point.
```

Comment:
```suggestion
In general, testing can be broken into levels of granularity. At the lowest level of granularity, we have unit tests, suitably written for testing the small chunks of code, generally functions and basic properties of structures. Climbing one level up, we have integration testing. Here, the focus is more on validation of correct software behaviour between inputs and outputs of functions. As the software written gets more complicated, systems level testing is introduced as the next level of granularity, which now is considered a black-box technique, usually tasked to be executed by software quality engineers instead of developers and hence we omit ourselves from the systems testing and higher level granularity at this point.
```